src/mmw/js/src/data_catalog/models.js

```diff
@@ -29,6 +29,23 @@ var Catalog = Backbone.Model.extend({
         error: '',
     },
 
+    searchIfNeeded: function(query, fromDate, toDate, bounds) {
+        var self = this,
+            isSameSearch = query === this.get('query') &&
+                           fromDate === this.get('fromDate') &&
+                           toDate === this.get('toDate') &&
+                           utils.formatBounds(bounds) === this.get('bbox');
+
+        if (!isSameSearch) {
```

Comment:
I will work out a way to explicitly cancel the ongoing promise for clarity.

I don't think, however, that `delete self.searchPromise` prevents the new promise from resolving correctly (and we don't use `this.searchPromise` anywhere else). 

This is not what I'd expect the behavior to be, but here's a little test I made: 
```js
var self = this;
var d1 = $.Deferred();

this.myPromise = $.when(d1)
       .done(function (v) {
           console.log(v);
       })
       .always(function() {
	   console.log("deleting in first");
           delete self.myPromise;
       });

var d2 = $.Deferred();

this.myPromise = $.when(d2)
        .done(function (v) {
            console.log(v)
        })
        .always(function() {
	   console.log("deleting in second");
           delete self.myPromise;
        });

d1.resolve( "first done" );
d2.resolve( "second done" );

/** outputs
VM10771:6 first done
VM10771:9 deleting in first
VM10771:17 second done
VM10771:20 deleting in second
**/
```